File: Core/Entities/Product.cs
namespace Core.Entities;

/// <summary>
/// Represents a product in the store, including details such as name, description, price, and stock quantity.
/// </summary>
public class Product : BaseEntity
{
  /// <summary>
  /// Gets or sets the name of the product.
  /// </summary>
  public required string Name { get; set; }

  /// <summary>
  /// Gets or sets the description of the product.
  /// </summary>
  public required string Description { get; set; }

  /// <summary>
  /// Gets or sets the price of the product.
  /// </summary>
  public decimal Price { get; set; }

  /// <summary>
  /// Gets or sets the URL of the product's picture.
  /// </summary>
  public required string PictureUrl { get; set; }

  /// <summary>
  /// Gets or sets the type or category of the product.
  /// </summary>
  public required string Type { get; set; }

  /// <summary>
  /// Gets or sets the brand of the product.
  /// </summary>
  public required string Brand { get; set; }

  /// <summary>
  /// Gets or sets the quantity of the product in stock.
  /// </summary>
  public int QuantityInStock { get; set; }
}

File: Client/src/app/shared/models/product.ts
/**
 * Interface representing a product in the system.
 */
export interface Product 
{
  /** The unique identifier for the product. */
  id: number;

  /** The name of the product. */
  name: string;

  /** A brief description of the product. */
  description: string;

  /** The price of the product. */
  price: number;

  /** The quantity of the product available in stock. */
  quantityInStock: number;

  /** The brand name of the product. */
  brand: string;

  /** The type or category of the product (e.g., 'hardware', 'peripheral'). */
  type: string;

  /** The URL to the image representing the product. */
  pictureUrl: string;
}